ksqldb-parser/src/main/java/io/confluent/ksql/parser/DefaultKsqlParser.java

```diff
@@ -47,6 +48,15 @@
           .map(DefaultKsqlParser::parsedStatement)
           .collect(Collectors.toList());
 
+    } catch (final ParsingException e) {
+      // ParsingException counts lines starting from 1
+      final String failedLine =  sql.split(System.lineSeparator())[e.getLineNumber() - 1];
+      // Since we return only the failed statement, the line number for it is always 1
```

Comment:
Sorry, but after thinking a little bit more, I think you had the right number in the previous commit. If a user submit > 1 statement, then they should know the line of the set of statements they submit. There are uses that call the REST API directly with several statements.

Could you go back to the original line number?

Reply:
Makes sense! I reverted the change in 441f6ba.